_episodes/36.md

```diff
@@ -0,0 +1,534 @@
+making operations faster than just bare metal speed that we are going to focus
+on.
+
+To demonstrate this, I'd like to use an analogy about how I think of this. 
+Comparing C++ and Java implementation is like comparing the two fastest men in 
+the world. Usain Bolt holds the most world records for mens track to this
+date, and teammate Yohan Blake holds many of the second place titles. Most of us
+know Usain Bolt is the fastest of the two, and you may not have known or
+remembered Yohan's name before. Want to hear something crazy, Yohan has beaten 
+Usain Bolt in a few races. The two are so close in speed, it's seconds to 
+milliseconds difference. The main difference in this analogy is that speed is
+the only thing that matters in an olymic race. Howver, programming languages and
+frameworks have a lot more tradeoffs.
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/36/usain-bolt-yohan-blake.webp"/>
+</p>
+
+The point is, Java is fast and more importantly, it abstracts a lot of burden
+maintaining and scaling out the code. This is conducive to a healthy open-source
+project and lowers the barrier for collaboration. Rather than go against this 
+and take on the feat of having to rewrite an entire system in C++, why not lean
+into the incredible innovation recent Java features have to offer to produce
+speed.
+
+Another important aspect is rather than chasing the fastest bare metal speed,
+it's also incredibly important to dedicate time into ensuring that Trino's
+optimizer is producing the best possible plans to avoid doing unnecessary work.
+To continue with the analogy, in a 100m race on a 400m track, imagine we have
+Usain and Yohan go head to head. We may expect that Usain will likely win, given
+his track record. However, if Usain is given the wrong instructions and runs in
+the wrong direction (300m), my bets are that Yohan will win the race. 
+
+In essence, the direction of Trino while still including bare metal performance
+improvements in the JVM, will instead focus on not wasting time with suboptimal
+query plans before or during runtime. There are so many optimizations that are
+constantly being added to every release that ultimately makes for a 
+work-smarter-not-harder query engine.
+
+## Concept of the episode: Java 17 and rearchitecting Trino
+
+As Trino prepares to [update to Java 17](https://github.com/trinodb/trino/issues/9876),
+we wanted to give a glimpse at what has happened between the current required
+JDK version, JDK 11, and future version JDK 17. Both of these versions are
+long-term support versions, and in the four years from 11 to 17 
+[a lot of exciting improvements were added.](https://openjdk.java.net/projects/jdk/17/jeps-since-jdk-11).
+
+### Java 17 updates
+
+Here are some [updates coming up in Java 17](https://openjdk.java.net/projects/jdk/17/jeps-since-jdk-11).
+
+#### Performance
+
+There were several JEPS that improve performance as well as many small changes
```

Comment:
Java Enhancement Proposals (JEP)

Reply:
also confirm that this is what JEP stands for

Reply:
actually it's JDK enhancement proposals